### PR TITLE
Alex aliyawer sprint1

### PR DESCRIPTION
**Pull request by rednaxela5950**
Tests and implementation of math.pow
Some tests fail. Need code review and help.

### CODE REVIEW

File: python/common/python/math.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package python;
+
+@org.python.Module(
+        __doc__ = "A module that supplies some mathematical functions.\n"+
+                    "Current functions:\n"+
+                    "pow(x, y) -> Return x raised to the power y as type float."
+)
+
+public class math extends org.python.types.Module {
```
Comment (ljbevemyr) [inline python/common/python/math.java:9]: Maybe change math to Math in order to follow the coding convention

File: python/common/python/math.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @org.python.Method(
+        __doc__ = "Returns x raised to the power y.\n",
+        args = {"x", "y"}
+    )
+
+    public static org.python.Object pow(org.python.Object x, org.python.Object y) {
+        double a;
+        double b;
```
Comment (ljbevemyr) [inline python/common/python/math.java:21]: Change the variable names a and b to something more explanatory.

File: python/common/python/math.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+public class math extends org.python.types.Module {
+    public math() {
+        super();
+    }
+
+    @org.python.Method(
+        __doc__ = "Returns x raised to the power y.\n",
+        args = {"x", "y"}
```
Comment (ljbevemyr) [inline python/common/python/math.java:16]: Maybe change variable names to something more explanatory, for example base and exponent.

File: python/common/python/math.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            b = ((org.python.types.Int) y).value;
+        } else if(y instanceof org.python.types.Float) {
+            b = ((org.python.types.Float) y).value;
+        } else {
+            throw new org.python.exceptions.TypeError("y needs to be of type int or float");
+        }
+
+        if(a < 0 || b < 0) {
+            throw new org.python.exceptions.TypeError("Only values of x >= 0 and y >= 0 are accepted");
+        }
```
Comment (ljbevemyr) [inline python/common/python/math.java:41]: To our understanding the python math pow should be able to take negative values for both the base and the exponent. Maybe look into this further.

Below is a part test of pow in a python3 shell
`
> import math
> pow(2, -3)
0.125
> pow(-2, 2)
4
> pow(-2, 2.5)
(1.7319121124709869e-15+5.656854249492381j)
>
 `

File: python/common/python/math.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    public math() {
+        super();
+    }
+
+    @org.python.Method(
+        __doc__ = "Returns x raised to the power y.\n",
+        args = {"x", "y"}
+    )
+
+    public static org.python.Object pow(org.python.Object x, org.python.Object y) {
```
Comment (ljbevemyr) [inline python/common/python/math.java:19]: Maybe the code could be divided up into multiple methods?

File: tests/stdlib/test_math.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import unittest
+
+from ..utils import TranspileTestCase
+
+
+class MathModuleTests(TranspileTestCase):
```
Comment (ljbevemyr) [inline tests/stdlib/test_math.py:6]: Maybe add some more edge case:
- Where the exponent is negative.
- Where both the base and the exponent is negative.
- When the output is a complex number.

**Review by ljbevemyr — CHANGES_REQUESTED** [review]
Good job! Followed most of the coding conventions, just som touchups :)
